app_server/app/adapters/socket/Room.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @module Room
+ * Abtraction of a chat room
+ */
+'use strict';
+
+var rfr = require('rfr');
+
+var Utility = rfr('app/util/Utility');
+
+var logger = Utility.createLogger(__filename);
+
+function Room(name, type) {
+  if (!name) {
+    logger.error('Room name is invalid');
+    throw new Error('Room name must be provided');
+  }
+
+  if (!type) {
+    logger.error('Room type is invalid');
+    throw new Error('Room type must be provided');
+  }
+
+  this.__name = name;
+  this.__type = type;
+  this.__clients = {};
+}
+
+var Class = Room.prototype;
+Room.ROOM_TYPES = Class.ROOM_TYPES = {
+  STREAM: 'stream',
+  GENERAL: 'general'
+};
```

Comment:
I'm not entirely confident on prototypes, but why should both the object and its prototype be set?